apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -110,7 +110,45 @@ def job_details
 
     hpc_job = project.job(job_details_params[:jobid].to_s, cluster_str)
 
-    render(partial: 'job_details', locals: { job: hpc_job })
+    @project = project
+
+    render(partial: 'job_details', locals: { job: hpc_job, project: @project })
+  end
+
+  # DELETE /projects/:project_id/jobs/:cluster/:jobid
+  def delete_job
+    @project = Project.find(job_details_params[:project_id])
+
+    cluster_str = job_details_params[:cluster].to_s
+
+    @project.remove_logged_job(job_details_params[:jobid].to_s, cluster_str)
+
+    @valid_project = Launcher.clusters?
+    @valid_scripts = Launcher.scripts?(@project.directory)
+    @scripts = Launcher.all(@project.directory)
+
+    render :show
+  end
+
+  # POST /projects/:project_id/jobs/:cluster/:jobid/stop
+  def stop_job
+    @project = Project.find(job_details_params[:project_id])
+    cluster_str = job_details_params[:cluster].to_s
+    cluster = OodAppkit.clusters[cluster_str.to_sym]
+
+    hpc_job = @project.job(job_details_params[:jobid].to_s, cluster_str)
+
+    begin
+      cluster.job_adapter.delete(job_details_params[:jobid].to_s) unless hpc_job.status.to_s == 'completed'
+    rescue StandardError => e
+      flash.now[:alert] = I18n.t('dashboard.jobs_project_generic_error', error: e.message.to_s)
+    end
+
+    @valid_project = Launcher.clusters?
+    @valid_scripts = Launcher.scripts?(@project.directory)
+    @scripts = Launcher.all(@project.directory)
+
+    render :show
```

Comment:
Similar comment to above, I think we should redirect instead of creating all these objects all over again.

Also seems like you need a `notice` in the case that it was successful.